python3-requirements.json

```diff
@@ -0,0 +1,143 @@
+                    "sha256": "66fa0c1e942478fd7970b791be017b1ca46556f54d44115008f3982e97ae0a9b"
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/70/8e/0e2d847013cb52cd35b38c009bb167a1a26b2ce6cd6965bf26b47bc0bf44/requests-2.31.0-py3-none-any.whl",
+                    "sha256": "58cd2187c01e70e6e26505bca751777aa9f2ee0b7f4300988b709f44e013003f"
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/61/b8/76d321077eca630b270b1ae03883936547a701a3ac8c91ad19216cf021cf/sepaxml-2.6.1-py3-none-any.whl",
+                    "sha256": "f110e4a11322c1ea46f86550d703468b109343b4cece1d5b792d394f03d86110"
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/a6/a5/c0b6468d3824fe3fde30dbb5e1f687b291608f9473681bbf7dabbf5a87d7/text_unidecode-1.3-py2.py3-none-any.whl",
+                    "sha256": "1311f10e8b895935241623731c2ba64f4c455287888b18189350b67134a822e8"
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/9b/81/62fd61001fa4b9d0df6e31d47ff49cfa9de4af03adecf339c7bc30656b37/urllib3-2.0.4-py3-none-any.whl",
+                    "sha256": "de7df1803967d2c2a98e4b11bb7d6bd9210474c46e8a0401514e3a42a75ebde4"
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/f4/24/2a3e3df732393fed8b3ebf2ec078f05546de641fe1b667ee316ec1dcf3b7/webencodings-0.5.1-py2.py3-none-any.whl",
+                    "sha256": "a0af1213f3c2226497a97e2b3aa01a7e4bee4f403f95be16fc9acd2947514a78"
+                },
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/28/19/17c655f3e7854e74003a225d0ada2c14944e08cf65bbdba9256b53ec77c7/xmlschema-2.4.0-py3-none-any.whl",
+                    "sha256": "dc87be0caaa61f42649899189aab2fd8e0d567f2cf548433ba7b79278d231a4a"
+                }
+            ]
+        },
+        {
+            "name": "python3-mt-940",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"mt-940\" --no-build-isolation"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "url": "https://files.pythonhosted.org/packages/49/86/c0baa3dcac450f784f776e5f245805da14c12f5caa26bfbb03710eb46582/mt_940-4.30.0-py2.py3-none-any.whl",
+                    "sha256": "66fa0c1e942478fd7970b791be017b1ca46556f54d44115008f3982e97ae0a9b"
+                }
+            ]
+        },
+        {
+            "name": "python3-cryptography",
```

Comment:
You need to:

1. Add the rust sdk to the manifest
2. Download the cryptography sdist tarball, extract and go to src/rust. Generate the cargo sources file with flatpak-cargo-generator (https://github.com/flatpak/flatpak-builder-tools). Then commit that file and set the necessary rust env vars for the build.
3. Add all build dependencies of cryptography: setuptools_rust (use flatpak pip generator from above to generate it)
4. python-cffi needs to built and installed before the cryptography build starts. Add it seperately before cryptography. Otherwise cryptography's hazmat bindings won't be build (but it's not a build time error, you'll see issues when running)

See https://github.com/flathub/io.github.hydrusnetwork.hydrus/blob/68e4120f51ae9411488fe46f853dc62820c97aed/python3-sources.json#L480-L555 as an example.

Reply:
Thanks!! I've added a locally tested version. This was a challenging task :)